src/EditorFeatures/CSharpTest/Formatting/CodeCleanupTests.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeCleanup;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Formatting
+{
+    [UseExportProvider]
+    public class CodeCleanupTests
+    {
+        [Fact]
+        [Trait(Traits.Feature, Traits.Features.CodeCleanup)]
+        public Task FormatDocumentRemoveUsings()
+        {
+            var code = @"using System;
+using System.Collections.Generic;
+class Program
+{
+    static void Main(string[] args)
+    {
+        Console.WriteLine();
+    }
+}
+";
+
+            var expected = @"using System;
+class Program
+{
+    static void Main(string[] args)
+    {
+        Console.WriteLine();
+    }
+}
+";
+            return AssertCodeCleanupResult(expected, code,
+                (CodeCleanupOptions.AreCodeCleanupRulesConfigured, enabled: true),
+                (CodeCleanupOptions.RemoveUnusedImports, enabled: true));
+        }
+
+        [Fact]
+        [Trait(Traits.Feature, Traits.Features.CodeCleanup)]
+        public Task FormatDocumentSortUsings()
+        {
+            var code = @"using System.Collections.Generic;
+using System;
+class Program
+{
+    static void Main(string[] args)
+    {
+        var list = new List<int>();
+        Console.WriteLine(list.Count);
+    }
+}
+";
+
+            var expected = @"using System;
+using System.Collections.Generic;
+class Program
+{
+    static void Main(string[] args)
+    {
+        var list = new List<int>();
+        Console.WriteLine(list.Count);
+    }
+}
+";
+            return AssertCodeCleanupResult(expected, code,
+                (CodeCleanupOptions.AreCodeCleanupRulesConfigured, enabled: true),
+                (CodeCleanupOptions.SortImports, enabled: true));
+        }
+
+        [Fact(Skip = "disable the test temporarily until figure out how to set up diagnostic analyzer")]
```

Comment:
this? http://source.roslyn.io/#Roslyn.Services.Test.Utilities/Diagnostics/TestDiagnosticAnalyzerService.cs,12